tests/test_windows.py

```diff
@@ -292,3 +292,17 @@ def test_intersection():
     """Window intersection works."""
     window = intersection(Window(0, 0, 10, 10), Window(8, 8, 12, 12))
     assert window == ((8, 10), (8, 10))
+
+
+def test_round_window_to_full_tiles():
+    with rasterio.open('tests/data/alpha.tif') as src:
+        block_shapes = src.block_shapes
+        test_window = ((321, 548), (432, 765))
```

Comment:
Can we also add tests for a window that is already at a block edge and a boundless window (going beyond the extent of the raster)?